hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileCreateRequest.java

```diff
@@ -154,9 +154,12 @@ public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
         .map(info -> info.getProtobuf(getOmRequest().getVersion()))
         .collect(Collectors.toList()));
 
+    KeyArgs resolvedArgs = resolveBucketAndCheckKeyAcls(newKeyArgs.build(),
+        ozoneManager, IAccessAuthorizer.ACLType.CREATE);
+
     generateRequiredEncryptionInfo(keyArgs, newKeyArgs, ozoneManager);
     CreateFileRequest.Builder newCreateFileRequest =
-        createFileRequest.toBuilder().setKeyArgs(newKeyArgs)
+        createFileRequest.toBuilder().setKeyArgs(resolvedArgs)
```

Comment:
Turns out this causes a regression where encrypted buckets won't correctly have file encryption info in keyargs set. Because encryption info is generated in `generateRequiredEncryptionInfo` above and put into `newKeyArgs`.

Credit to @jojochuang for locating this bug.

Reply:
HDDS-10363